exchanges/okx/okx.go

```diff
@@ -54,404 +56,81 @@ const (
-	errMissingRequiredParameter                = errors.New("missing required parameter")
-	errMissingMakerInstrumentSettings          = errors.New("missing maker instrument settings")
-	errInvalidSubAccountName                   = errors.New("invalid sub-account name")
-	errInvalidAPIKey                           = errors.New("invalid api key")
-	errInvalidAlgoID                           = errors.New("invalid algo id")
-	errInvalidMarginTypeAdjust                 = errors.New("invalid margin type adjust, only 'add' and 'reduce' are allowed")
-	errInvalidAlgoOrderType                    = errors.New("invalid algo order type")
-	errEmptyArgument                           = errors.New("empty argument")
-	errInvalidIPAddress                        = errors.New("invalid ip address")
-	errInvalidAPIKeyPermission                 = errors.New("invalid API Key permission")
-	errInvalidResponseParam                    = errors.New("invalid response parameter, response must be non-nil pointer")
-	errEmptyPlaceOrderResponse                 = errors.New("empty place order response")
-	errTooManyArgument                         = errors.New("too many cancel request params")
-	errIncompleteCurrencyPair                  = errors.New("incomplete currency pair")
-	errInvalidDuration                         = errors.New("invalid grid contract duration, only '7D', '30D', and '180D' are allowed")
-	errInvalidProtocolType                     = errors.New("invalid protocol type, only 'staking' and 'defi' allowed")
-	errExceedLimit                             = errors.New("limit exceeded")
-	errOnlyThreeMonthsSupported                = errors.New("only three months of trade data retrieval supported")
-	errOnlyOneResponseExpected                 = errors.New("one response item expected")
-	errNoInstrumentFound                       = errors.New("no instrument found")
 )
 
 /************************************ MarketData Endpoints *************************************************/
 
+var orderTypeFromStringMap = map[string]order.Type{OkxOrderMarket: order.Market, OkxOrderLimit: order.Limit, OkxOrderPostOnly: order.PostOnly, OkxOrderFOK: order.FillOrKill, OkxOrderIOC: order.ImmediateOrCancel, OkxOrderOptimalLimitIOC: order.OptimalLimitIOC}
+
 // OrderTypeFromString returns order.Type instance from string
 func (ok *Okx) OrderTypeFromString(orderType string) (order.Type, error) {
-	switch strings.ToLower(orderType) {
-	case OkxOrderMarket:
-		return order.Market, nil
-	case OkxOrderLimit:
-		return order.Limit, nil
-	case OkxOrderPostOnly:
-		return order.PostOnly, nil
-	case OkxOrderFOK:
-		return order.FillOrKill, nil
-	case OkxOrderIOC:
-		return order.ImmediateOrCancel, nil
-	case OkxOrderOptimalLimitIOC:
-		return order.OptimalLimitIOC, nil
-	default:
-		return order.UnknownType, fmt.Errorf("%w %v", errInvalidOrderType, orderType)
+	orderTypeValue, okay := orderTypeFromStringMap[strings.ToLower(orderType)]
```

Comment:
suggestion: maps on average with such a small data set are less performant than the switch case, so you can revert but it will be faster to iterate over a slice instead if you want to keep this declared variable and small concise code. Up to you. It's not a big deal you are only saving a couple ns.